src/server/JasmineGraphInstanceService.cpp

```diff
@@ -80,4301 +144,5243 @@ void *instanceservicesession(void *dummyPt) {
+            workerThreads[count] =
+                std::thread(&JasmineGraphInstanceService::collectTrainedModelThreadFunction, sessionargs, hostName,
+                            workerPartitions.port, workerPartitions.dataPort, graphID, *it);
+            count++;
+        }
+    }
 
-            bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            string graphVertexCount = (data);
-            graphVertexCount = Utils::trim_copy(graphVertexCount, " \f\n\r\t\v");
-            instance_logger.log("Received Graph ID:" + graphID + " Vertex Count: " + graphVertexCount, "info");
+    for (int threadCount = 0; threadCount < count; threadCount++) {
+        workerThreads[threadCount].join();
+        std::cout << "Thread [C]: " << threadCount << " joined" << std::endl;
+    }
+}
 
-            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
-            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::OK, "info");
+int JasmineGraphInstanceService::collectTrainedModelThreadFunction(instanceservicesessionargs *sessionargs,
+                                                                   std::string host, int port, int dataPort,
+                                                                   std::string graphID, std::string partition) {
+    bool result = true;
+    std::cout << pthread_self() << " host : " << host << " port : " << port << " DPort : " << dataPort << std::endl;
+    int sockfd;
+    char data[INSTANCE_DATA_LENGTH + 1];
+    bool loop = false;
+    socklen_t len;
+    struct sockaddr_in serv_addr;
+    struct hostent *server;
 
-            bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            string alphaValue = (data);
-            alphaValue = Utils::trim_copy(alphaValue, " \f\n\r\t\v");
-            instance_logger.log("Received alpha: " + alphaValue, "info");
+    sockfd = socket(AF_INET, SOCK_STREAM, 0);
 
-            double alpha = std::stod(alphaValue);
+    if (sockfd < 0) {
+        std::cerr << "Cannot create socket" << std::endl;
+        return 0;
+    }
 
-            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
-            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::OK, "info");
+    if (host.find('@') != std::string::npos) {
+        host = Utils::split(host, '@')[1];
+    }
 
-            bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            string iterationsValue = (data);
-            iterationsValue = Utils::trim_copy(iterationsValue, " \f\n\r\t\v");
-            instance_logger.log("Received iteration count: " + iterationsValue, "info");
+    server = gethostbyname(host.c_str());
+    if (server == NULL) {
+        std::cerr << "ERROR, no host named " << server << std::endl;
```

Comment:
Lets move this to a log message

Reply:
Replaced with `instance_logger.error("ERROR, no host named " + host);` in next PR

Reply:
Ok